lib/ui/painting.dart

```diff
@@ -2965,7 +2965,7 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// multiple draw commands intersect with the clip boundary, this can result
   /// in incorrect blending at the clip boundary. See [saveLayer] for a
   /// discussion of how to address that and some examples of using [clipRRect].
-  void clipRRect(RRect rrect, [bool doAntiAlias = true]) {
+  void clipRRect(RRect rrect, {bool doAntiAlias = true}) {
```

Comment:
i was just about to comment that it's weird that changing this didn't affect any of the documentation, when i noticed that it's actually worse, because the documentation contradicts the code now. Why is Paint.isAntiAlias (what the docs say control this) not sufficient?

Reply:
also why is this not applied to clipRect as well?

Reply:
@Hixie : Ah, I forgot to update the comments. Thanks for spotting that!

As for clipRect, I didn't intend to introduce more nobs for it because we never had saveLayer for clipRect, and clipRect is much cheaper than clipRRect and clipPath. Do you want to introduce Clip.hardEdge, Clip.antiAlias, and Clip.antiAliasWithSaveLayer to clipRect as well? I can certainly do that if it makes sense.

Reply:
This patch isn't about layers, it's about whether to antialias.

I don't see why clipRRect and clipRect would be different.

But as I comment above, the docs say Paint.isAntiAlias is the solution here, so, isn't it?

Reply:
@Hixie the doc is clearly out of sync for a long time. The source code is doing anti-aliased clipping all the time regardless of what `Paint.isAntiAlias` is: https://github.com/flutter/engine/pull/5638/files#diff-36581c173eef0311bb48263255854c24L172.

Reply:
well then we should definitely update the docs. :-)